linode/manager
# upcoming: [M3-8217] - Tax Id Notifications & Icon

## Description 📝
New notifications for Tax Id

## Changes  🔄
- Add two new notifications when tax id is valid/invalid
- Add warning icon next to tax id in billing section
   - Changed icon to solid variant for visibility (only used in this location in the app) 

## Target release date 🗓️
7/8

## Preview 📷

https://github.com/linode/manager/assets/125309814/b692dd1b-04d3-45d3-ac17-15fc03dfdea9


| Desc  | SS   |
| ------- | ------- |
| Notifications | ![Screenshot 2024-06-26 at 8 56 23 AM](https://github.com/linode/manager/assets/125309814/882971d0-9326-4b4a-8485-c2bad720998c) |
| Toast | ![Screenshot 2024-07-10 at 12 09 48 PM](https://github.com/linode/manager/assets/125309814/28d9d489-a1ad-4de9-ae17-9018e6a9a1b0) |
| Event | ![Screenshot 2024-07-10 at 3 04 59 PM](https://github.com/linode/manager/assets/125309814/da976ef2-6b0d-4356-b11d-e9bb1c8240bf) |

| Icon Before  | Icon After (ignore dark mode color changes) |
| ------- | ------- |
|![Screenshot 2024-06-07 at 12 57 34 PM](https://github.com/linode/manager/assets/125309814/26ce5c17-1727-4412-9ebb-5b685313492b)| ![Screenshot 2024-07-08 at 9 59 11 AM](https://github.com/linode/manager/assets/125309814/ce821dbb-f2f4-4539-b55b-2141f3767697)|

## How to test 🧪

### Prerequisites
- If you want to test the events part of this, you'll need to setup DevEnv (see me about details)
- Otherwise, you can locally comment out the rendering conditions for the TooltipIcon

### Validation
- Checkout this branch in devenv
- Checkout latest develop branch of apinext in devenv
- Add a row to the TaxValidation table with format_valid set to 1, and tax_registered to yes.
- curl /account/notifications and notice the tax_id_valid notification.
- Set format_valid to 0, and curl /account/notifications again, notice tax_id_invalid notification. Try again with format_valid set to 1, and tax_registered set to no.

## Related
- https://github.com/linode/manager/pull/10664
